### DISCUSSION TITLE
post/dapr-cron-binding

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#15] [Announcements]
# post/dapr-cron-binding

Mark Heath's Development Blog

https://markheath.net/post/dapr-cron-binding

**Comment by yuriy-butkevych** [comment]
Thank you for the great article, Mark!
Did you encounter with the problem that the “cron binding” triggers API request on each instance of the scoped component? I am expecting that just one instance will handle the request. Thank you

**Reply by markheath** [reply]
Great question - I hadn't tried it out in that scenario, and I had assumed it would behave similar to Azure Functions scheduled jobs, which uses a blob lease to ensure only one instance performs the scheduled action. However, I did a bit of digging and found these issues: 
https://github.com/dapr/components-contrib/issues/1927
https://github.com/dapr/components-contrib/issues/1249#issuecomment-959799552

Which suggest that all instances will get the message. I guess there are situations in which that's maybe what you want but it would be nice if Dapr offered both modes. And given that there is a distributed lock building block in Dapr, it shouldn't be too hard for them to implement.
